### DISCUSSION TITLE
Release Notes not showing up in Renovate pull requests

### DISCUSSION THREAD

**Discussion by AswaniJB** [renovatebot/renovate#16211] [Request Help]
We are using renovate bot for dependency management. The pull requests get created automatically but the release notes are not populated. We would like to see the changes between the changes part of the new package/release.

Has anyone tried/achieved this?  Thanks much!

**Comment by rarkins** [comment]
I have achieved this

**Reply by AmyScript** [reply]
@rarkins we are running our own instance of renovate and we are using our own instance of Github.  Is there anything special that needs to be done to get the release notes?  We have tried updating the GITHUB_COM_TOKEN but still doesn't work.  Thanks for your help.

**Reply by AmyScript** [reply]
We have also updated the renovate config to have `"fetchReleaseNotes": true` and added CHANGELOG.md files to the root of the repos.

**Reply by rarkins** [reply]
Do you get the release notes for *any* packages, e.g. public ones? Or you have a complete absence of release notes for all packages - both private and public?

**Reply by AswaniJB** [reply]
We have complete absence of release notes for all packages - public and private.

**Reply by rarkins** [reply]
Do you use a private registry for all packages, or a mix of public and private? And how confident are you that you have your GITHUB_COM_TOKEN right?

**Reply by AswaniJB** [reply]
All the packages are from a private nexus registry, We mirror packages from https://www.npmjs.com/.

**Reply by AswaniJB** [reply]
We are confident in the token. It looks correct.

**Reply by rarkins** [reply]
Please create a test repo with a single dependency which you know displays a changelog when hosted on GitHub.com. For example npm package chalk@5.0.0, which should have an update pending to 5.0.1.

Now run renovate at debug logging mode for just that repo and capture the entire log to include here (assuming it creates a PR for 5.0.1 without changelog as expected)

**Reply by AswaniJB** [reply]
We tried below steps:
1. We created a sample repo and used a library which has a version update with release notes.
2. We used our enterprise Github token ( the repo and the lib resides inside enterprise Github).
3. We still don't see the release notes in the PR. Attaching the [logs ](https://github.com/renovatebot/renovate/files/9161518/Renovate.log)
(from entering the webhook event till end). Would you be able to check the logs and let us know what could have gone wrong? Thanks!

**Reply by rarkins** [reply]
Step 2 is incorrect, I said to use a public package you know works on GitHub.com

**Reply by AswaniJB** [reply]
Thanks for the response rarkins, we did try with public packages, and they work fine.

Most of our packages are in enterprise GitHub, we would like to have release notes for those packages too.

**Reply by rarkins** [reply]
You wrote earlier "We have complete absence of release notes for all packages - public and private.". Was that incorrect, or did you take steps to fix it?

**Reply by AswaniJB** [reply]
When we used the right GitHub token for the public packages, we got release notes for them.

**Reply by rarkins** [reply]
So now pick a single package which you think should have release notes. What type of package is it and where is it hosted? Eg npm package hosted on artifactory?

**Reply by AswaniJB** [reply]
@rarkins: We noticed we are using an old version of renovate (1.5.0) and decided to try this by upgrading to a stable version. Will keep you posted. Thanks much!

**Reply by AswaniJB** [reply]
@rarkins : We updated the version of Renovate to the latest and we are able to successfully fetch the release notes for the public packages, but for the internal packages residing in Enterprise GitHub, we still don't see the release notes. Is there a specific config to be turned on in Renovate, to get the release notes for internal ones?

**Reply by rarkins** [reply]
No, there's no magic setting to get them working. Create a test repo with one dependency, one PR, and non-working release notes and capture the entire debug log.

**Reply by AswaniJB** [reply]
Sure will share the logs. Just wanted to confirm if it matters if the enterprise GitHub resides on-premises.
